presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPruneDistinctOverInPredicate.java

```diff
@@ -0,0 +1,87 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.sql.planner.Symbol;
+import io.prestosql.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.prestosql.sql.planner.iterative.rule.test.PlanBuilder;
+import io.prestosql.sql.planner.plan.PlanNode;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.semiJoin;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.values;
+import static io.prestosql.sql.planner.iterative.rule.test.PlanBuilder.expression;
+
+public class TestPruneDistinctOverInPredicate
+        extends BaseRuleTest
+{
+    @Test
+    public void test()
+    {
+        tester().assertThat(new PruneDistinctOverInPredicate())
+                .on(TestPruneDistinctOverInPredicate::semiJoinWithDistinctAsFilteringSource)
+                .matches(
+                        semiJoin("leftKey", "rightKey", "match",
+                                values("leftKey"),
+                                values("rightKey")));
+    }
+
+    @Test
+    public void testDoesNotFire()
+    {
+        tester().assertThat(new PruneDistinctOverInPredicate())
+                .on(TestPruneDistinctOverInPredicate::semiJoinWithAggregationAsFilteringSource)
+                .doesNotFire();
+    }
+
+    private static PlanNode semiJoinWithDistinctAsFilteringSource(PlanBuilder p)
+    {
+        Symbol match = p.symbol("match");
+        Symbol leftKey = p.symbol("leftKey");
+        Symbol rightKey = p.symbol("rightKey");
+        return p.semiJoin(
+                leftKey,
+                rightKey,
+                match,
+                Optional.empty(),
+                Optional.empty(),
+                p.values(leftKey),
+                p.aggregation(builder -> builder
+                        .singleGroupingSet(rightKey)
+                        .source(p.values(rightKey))));
+    }
+
+    private static PlanNode semiJoinWithAggregationAsFilteringSource(PlanBuilder p)
+    {
+        Symbol match = p.symbol("match");
+        Symbol leftKey = p.symbol("leftKey");
+        Symbol rightValue = p.symbol("rightValue");
+        Symbol rightKey = p.symbol("rightKey");
+        return p.semiJoin(
+                leftKey,
+                rightKey,
+                match,
```

Comment:
Inline variables that are used only once

Reply:
done